packages/api/lib/writeRecords/write-granules.js

```diff
@@ -429,6 +430,85 @@ const _writeGranule = async ({
   await publishGranuleSnsMessageByEventType(granuletoPublish, snsEventType);
 };
 
+/**
+ * Update granule record status to queued in DynamoDB, PostgreSQL, Elasticsearch.
+ * Publish SNS event for updated granule.
+ *
+ * @param {Object}  params
+ * @param {Object}  params.apiGranuleRecord      - API Granule object to write to
+ *                                                 the database
+ * @param {Object}  params.originalApiGranule    - API Granule object before status update
+ * @param {Object}  params.postgresGranuleRecord - PostgreSQL granule record to write
+ *                                                 to database
+ * @param {string}  params.granuleId             - Granule ID
+ * @param {Object}  params.granuleModel          - Instance of DynamoDB granule model
+ * @param {Object}  params.granulePgModel        - @cumulus/db compatible granule module instance
+ * @param {Knex}    params.knex                  - Knex object
+ * @param {string}  params.snsEventType          - SNS Event Type, defaults to 'Update'
+ * @param {Object}  params.esClient              - Elasticsearch client
+ * returns {Promise}
+ */
+const _writeQueuedGranule = async ({
+  apiGranuleRecord,
+  originalApiGranule,
+  postgresGranuleRecord,
+  granuleId,
+  granuleModel,
+  granulePgModel,
+  knex,
+  snsEventType = 'Update',
+  esClient,
+}) => {
+  const apiGranule = apiGranuleRecord;
+  delete apiGranule.execution;
+
+  let granuleCumulusId;
+  const status = 'queued';
+  await createRejectableTransaction(knex, async (trx) => {
+    granuleCumulusId = await granulePgModel.update(
+      trx,
+      { cumulus_id: postgresGranuleRecord.cumulus_id },
+      { status }
+    );
+    log.info(`Successfully wrote granule ${granuleId} to PostgreSQL`);
+    try {
+      // delete execution field
+      await granuleModel.update({ granuleId }, { status }, ['execution']);
+      log.info(`Successfully wrote granule ${granuleId} to DynamoDB`);
+      await upsertGranule({
+        esClient,
+        updates: apiGranule,
+        index: process.env.ES_INDEX,
+      });
+      log.info(`Successfully wrote granule ${granuleId} to Elasticsearch`);
+    } catch (writeError) {
+      log.info(`Writes to DynamoDB/Elasticsearch failed, rolling back all writes for granule ${apiGranuleRecord.granuleId}`);
+      // On error, change the Dynamo record back to original status to ensure that all systems
+      // stay in sync
+      await granuleModel.update({ granuleId }, {
+        status: originalApiGranule.status,
+        execution: originalApiGranule.execution,
+      });
```

Comment:
```suggestion
      await granuleModel.create(apiGranule);
```

This should just revert the record back to what it was originally without having to specify individual fields

Reply:
[`44726d8` (#2593)](https://github.com/nasa/cumulus/pull/2593/commits/44726d85a533a55e7a6fe66f4b38da371c329862)